### ISSUE TITLE
Concourse's vault client does a ton of SRV lookups

### ISSUE THREAD

**Issue by hoegaarden** [concourse/concourse#8304]
## Steps to reproduce

- Deploy concourse with the vault integration
- configure the vault URL without the port, e.g.: `https://vault.enterprise.tld`
- ensure you have no SRV record for `_http._tcp.vault.enterprise.tld`
- check the DNS SRV requests that go out from concourse / hit your DNS server

=> you should see a ton of requests, all of them not returning anything
=> concourse is, however, still working correcly

## Expected results

When concourse runs, it doesn't do too many (sorry for being vague here) SRV lookups.

## Actual results

We see a lot of SRV lookups, and they all fail, because we don't have the SRV records in DNS.

On one concourse we've seen ~14 of those lookups per second (would amount to ~1.2m per day); and we have about 10ish conci around.

## Additional context

We have concourse talking to vault and we've set the vault URL to `https://vault.enterprise.tld` (note: we don't specify the port explicitly here).
We see a lot of DNS SRV queries for `_http._tcp.vault.enterprise.tld` which fail.
Everything still works, but the team that manages the resolvers is not happy about us, rather our conci, hammering their systems.

We think this is related to [this issue](https://github.com/hashicorp/vault/issues/5540), which shows, that if the vault URL has not explicitly specified a port, a SRV request is issued, not matter what.
There is a new version of the package (`github.com/hashicorp/vault/api@v1.5.0`) which still seems to show the same behaviour, but gives us a switch (`VAULT_SRV_LOOKUP`) to turn of SRV lookups.

I am not sure what we can / should do about that, but some thoughts:
- In the [concourse docs](https://concourse-ci.org/vault-credential-manager.html) explicitly call out, that vault will do SRV lookups when there is no port in the vault URL
- Update to `github.com/hashicorp/vault/api@v1.5.0` and document the option to dsiable SRV requests via `VAULT_SRV_LOOKUP`

## Triaging info

* Concourse version: `v7.4.0` -- but we think pretty much any concourse with the same vault package will show that behaviour
* Did this used to work? Didn't check if there was different behaviour in older versions of `github.com/hashicorp/vault/api` :shrug:

**Comment by xtremerui** [comment]
> There is a new version of the package (`github.com/hashicorp/vault/api@v1.5.0`) which still seems to show the same behaviour, but gives us a switch (`VAULT_SRV_LOOKUP`) to turn of SRV lookups.

According to this [commit](https://github.com/hashicorp/vault/commit/8dd6180e313c43ee6e2302dcf9f8cf5cd8705765), it will do the SRV lookup only when the env var is configured. I am wondering why you need to set `VAULT_SRV_LOOKUP ` to disable it.

**Comment by taylorsilva** [comment]
Sounds like upgrading the package will work fine then. Their code indicates this is still true, that SRV lookups dont happen by default anymore: https://github.com/hashicorp/vault/blob/b89d85779063c3701436d26101d3f8c540543f4c/api/client.go#L1115-L1123

**Comment by hoegaarden** [comment]
Yeah, I slightly misread the code of the newer vault api version, I think. It should now only do an SRV lookup if
- vault's URL does **not** explicitly specify a port
- **and** the `VAULT_SRV_LOOKUP` is explicitly set to `true`

So I think that would be good standard behaviour for concourse.

If the docs should should call out, that you can enable SRV lookups by setting `VAULT_SRV_LOOKUP`, is a thing I leave to you all.
One could consider that a breaking change, though, because concourse would silently not do the SRV lookups anymore if the URL does not sepcify a port.
